test/migration/setup_runner.go

```diff
@@ -185,6 +194,15 @@ func (r *SetupMigrationRunner) prepareBannedUser(t *testing.T) {
 	_, err = hostAwait.WithRetryOptions(wait.TimeoutOption(time.Second*15)).WaitForUserSignup(t, userSignup.Name,
 		wait.ContainsCondition(wait.Banned()[0]))
 	require.NoError(t, err)
+
+	// verify that space (and space binding) is deleted, to be sure that the space counter is decremented
+	err = hostAwait.WaitUntilSpaceAndSpaceBindingsDeleted(t, userSignup.Status.CompliantUsername)
+	require.NoError(t, err)
+
+	// let's wait until ToolchainStatus is updated with the latest numbers from the space counter
+	_, err = hostAwait.WaitForToolchainStatus(t, wait.UntilToolchainStatusUpdatedAfter(time.Now()))
```

Comment:
Same as above, I think this can potentially race. Is there a way of knowing e.g. the specific number of spaces that we should expect?